landside/catkin_ws/src/gui/src/gui/controls_widget.py

```diff
@@ -0,0 +1,267 @@
+from python_qt_binding import loadUi
+from python_qt_binding.QtWidgets import QWidget
+from python_qt_binding.QtCore import QTimer
+
+import rospy
+import resource_retriever as rr
+from tf.transformations import euler_from_quaternion, quaternion_from_euler
+import dynamic_reconfigure.client
+import numpy as np
+import rosservice
+
+from std_srvs.srv import SetBool
+from geometry_msgs.msg import Pose, Twist, Point, Quaternion, Vector3
+
+from gui.pose_twist_dialog import PoseTwistDialog
+from gui.pid_dialog import PidDialog
+import math
+
+
+class ControlsWidget(QWidget):
+
+    DIRS = ['x', 'y', 'z', 'roll', 'pitch', 'yaw']
+    PID = ['p', 'i', 'd']
+
+    def __init__(self):
+
+        super(ControlsWidget, self).__init__()
+
+        ui_file = rr.get_filename('package://gui/resource/ControlsWidget.ui', use_protocol=False)
+        loadUi(ui_file, self)
+
+        self.setObjectName('ControlsWidget')
+
+        self.background_colors = {
+            "green": "background-color: #8aff92",
+            "red": "background-color: #ff7878"
+        }
+        self.controls_button_text = {
+            "enable": "Enable Controls",
+            "disable": "Disable Controls"
+        }
+        self.enable_controls_button.clicked.connect(self.enable_state_clicked)
+        self.enable_controls_button.setText(self.controls_button_text["enable"])
+        self.enable_controls_button.setStyleSheet(self.background_colors["green"])
+
+        self.enable_controls_timer = QTimer(self)
+        self.enable_controls_timer.timeout.connect(self.check_enable_controls)
+        self.enable_controls_timer.start(100)
+
+        self.state_times = {
+            'pose': rospy.Time.now(),
+            'twist': rospy.Time.now(),
+            'power': rospy.Time.now()
+        }
+
+        self.desired_pose_sub = rospy.Subscriber('controls/desired_pose', Pose, self.pose_received)
+        self.desired_twist_sub = rospy.Subscriber('controls/desired_twist', Twist, self.twist_received)
+        self.desired_power_sub = rospy.Subscriber('controls/desired_power', Twist, self.power_received)
+
+        self.gui_publishing = False
+        self.state_timer = QTimer(self)
+        self.state_timer.timeout.connect(self.check_times)
+        self.state_timer.setInterval(100)
+        self.state_timer.start()
+        self.publishing_timer = QTimer(self)
+        self.publishing_timer.timeout.connect(self.publish_controls)
+        self.publishing_timer.setInterval(100)
```

Comment:
This rate is currently too slow for controls. I'm assuming this means the desired state is being published at 10Hz. This is the theoretical refresh rate for controls but timing mismatches cause controls to disable and reenable after every publishing event. Consider bumping this to around 15 Hz